package/metrics_daemon/metrics_daemon/src/metrics_daemon.c

```diff
@@ -0,0 +1,375 @@
+/*
+ * Copyright (C) 2018 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include <getopt.h>
+#include <string.h>
+#include <unistd.h>
+#include <ftw.h>
+#include <json-c/json.h>
+#include <libpiksi/logging.h>
+#include <libpiksi/settings.h>
+#include <libpiksi/util.h>
+#include <libpiksi/settings.h>
+#include "sbp.h"
+
+#define PROGRAM_NAME "metrics_daemon"
+
+#define METRICS_ROOT_DIRECTORY "/var/log/metrics"
+#define METRICS_OUTPUT_FILENAME "/var/log/metrics.json"
+#define METRICS_USAGE_UPDATE_INTERVAL_MS (10000u)
+
+#define MAX_FOLDER_NAME_LENGTH 64
+static json_object *jobj_root = NULL;
+static json_object *jobj_cur = NULL;
+static char *root_name = NULL;
+static unsigned int root_length = 0;
+static bool first_folder = true;
+static char *target_file = METRICS_OUTPUT_FILENAME;
+const char *metrics_path = METRICS_ROOT_DIRECTORY;
+static bool enable_log_to_file = false;
+
+
+int handle_walk_path(const char *fpath, const struct stat *sb, int tflag);
+char *extract_filename(const char *str);
+json_object *init_json_object(const char *path);
+
+/**
+ * @brief function updates json to file
+ *
+ * @param root       pointer to the root of json object
+ * @param file_name  file to write to
+ */
+static int write_json_to_file(struct json_object *root, const char *file_path)
+{
+  return file_append_string(file_path, json_object_to_json_string(root));
+}
+
+/**
+ * @brief static function that returns file name from path
+ * Ref: Updated code from
+ * https://cboard.cprogramming.com/c-programming/109469-scan-filename-make-list.html
+ *
+ * @param  str        file path in ASCII
+ * @return file_name  file name
+ */
+char *extract_filename(const char *str)
```

Comment:
Is there any reason we can't use the `basename` function?  See: https://linux.die.net/man/3/basename -- particularly if we use the GNU version of basename (at the bottom of the link above) we can avoid allocating memory, and just return (or use) a `const char*`.